torch/serialization.py

```diff
@@ -553,6 +562,36 @@ def maybe_decode_ascii(bytes_str):
             return bytes_str.decode('ascii')
         return bytes_str
 
+    deserialized_storage_keys = []
+    pending_tensors = []
+
+    tensor_data_pos = []
+
+    def read_from_file(f, root_key, recursing=False):
+        # Read the tensor from the file if it hasn't alread been read,
+        # don't do anything if this tensor has already been loaded from the
+        # file
+        if root_key not in initialized_deserialized_objects:
```

Comment:
you don't need this logic any more, do you? just unconditionally jump to the offset and back?